tests/testutils/goroutineassistant.go

```diff
@@ -18,23 +18,25 @@ import (
 	"fmt"
 	"sync"
 	"testing"
+
+	"github.com/coreos/rkt/Godeps/_workspace/src/github.com/steveeJ/gexpect"
 )
 
 type GoroutineAssistant struct {
-	s  chan string
+	s  chan error
 	wg sync.WaitGroup
 	t  *testing.T
 }
 
 func NewGoroutineAssistant(t *testing.T) *GoroutineAssistant {
 	return &GoroutineAssistant{
-		s: make(chan string),
+		s: make(chan error),
 		t: t,
 	}
 }
 
 func (a *GoroutineAssistant) Fatalf(s string, args ...interface{}) {
-	a.s <- fmt.Sprintf(s, args...)
+	a.s <- fmt.Errorf(s, args...)
```

Comment:
Two of three issues I wrote before are still not addressed.

- This function is not supposed to return - it should call `runtime.Goexit()` as the last thing in its body. We assume this function does not return in other places (we removed `return` clauses).
- I think that `a.s` channel should be non-blocking (as in `s: make(chan error, 10)`).